File: src/main/java/com/sasfmlzr/apivk/object/UserRights.java
package com.sasfmlzr.apivk.object;


import java.util.Objects;

public class UserRights {


    private Boolean allowWriteToBot=false;        // разрешение писать боту
    private Boolean adminCommands=false;          // разрешение на админские команды
    private String nameRight;


    public UserRights(String nameRight) {
        if (Objects.equals(nameRight, "Бог")){
            this.adminCommands=true;
            this.allowWriteToBot=true;
        }
        if (Objects.equals(nameRight, "Пользователь")){
            this.adminCommands=false;
            this.allowWriteToBot=true;
        }
        if (Objects.equals(nameRight, "Колян")){
            this.adminCommands=false;
            this.allowWriteToBot=false;
        }
        this.nameRight = nameRight;
    }


    public UserRights() {
    }


    public String getNameRight()
    {
        return nameRight;
    }

    public Boolean getAllowWriteToBot()
    {
        return allowWriteToBot;
    }

    public Boolean getAdminCommands()
    {
        return adminCommands;
    }



}

File: src/main/java/com/sasfmlzr/apivk/object/UserRights.kt
package com.sasfmlzr.apivk.`object`

import java.util.Objects

class UserRights(val nameRight: String) {

    var allowWriteToBot: Boolean? = false
        private set        // разрешение писать боту
    var adminCommands: Boolean? = false
        private set          // разрешение на админские команды


    init {
        if (nameRight == "Бог") {
            this.adminCommands = true
            this.allowWriteToBot = true
        }
        if (nameRight == "Пользователь") {
            this.adminCommands = false
            this.allowWriteToBot = true
        }
        if (nameRight == "Колян") {
            this.adminCommands = false
            this.allowWriteToBot = false
        }
    }
}
